src/emmy/tape.cljc

```diff
@@ -477,13 +543,12 @@
         (s/mapr #(->partials % tag) output)
 
         (f/function? output)
+        (->partials-fn output tag)
 
-        ;; TODO this needs to handle perturbation confusion with tag
-        ;; replacement. Make something similar to extract-tangent-fn and test
-        ;; this so we don't allow an amazing bug.
-        (comp #(->partials % tag) output)
+        (v/scalar? output)
+        (->Completed {})
 
-        :else (->Completed {})))
+        :else (u/unsupported "Output not handled yet!")))
```

Comment:
I want to replace this `cond` with something better, but I'm still thinking about it...